### PR TITLE
game logic complete

### PR DESCRIPTION
**Pull request by GEvanscsumb**
finished most of the room logic. game can run almost completely in the loop. need help finding a way to instance a room specific fight within that room's class.

### CODE REVIEW

**Comment by tramstheman** [comment]
This all looks good, my first question is does everything run as expected?

File: GameEngine/BossRoom.py [hunk 1]
```diff
@@ -5,5 +5,5 @@
 class BossRoom(Room):
   def __init__(self):
     Room.__init__(self)
-    self.image = Utils.loadImage("/rock_cave1.png")
-    self.scripts = ["preBoss.txt", "nochzBoss.txt", "chzBoss.txt", "excalPre.txt", "excalBoss.txt", "postBoss.txt", "badend.txt", "goodend.txt"]
+    self.image = Utils.loadImage("/dungeon_master_mark.png")
```
Comment (tramstheman) [inline GameEngine/BossRoom.py]: probably not important for this release, due to time, but I am wondering how much sense it makes to load a character sprite as the room image?

File: GameEngine/Entrance.py [hunk 2]
```diff
@@ -8,8 +8,9 @@ def __init__(self):
     self.image = Utils.loadImage("/rock_cave2.png")
     self.scripts = ["entranceroom.txt", "skelfirst.txt", "entwcrystal.txt"]
     skeleton = Enemy("Skeleton", 3)
+    skeleton.hp = 100
     skeleton.strength += 3
-    skeleton.dexterity = 1
+    skeleton.dexterity = 0
```
Comment (tramstheman) [inline GameEngine/Entrance.py]: was 1 dexterity having an issue?

File: GameEngine/Utils.py [hunk 3]
```diff
@@ -47,6 +47,9 @@ def battleSequence(player, entity):
             if player.isAlive():
                 if userInput == "a":
                     battle(player, entity)
+                    if not entity.isAlive():
+                        showInformation("Dan has slain the enemy!")
+                        player.levelUp()
```
Comment (tramstheman) [inline GameEngine/Utils.py]: Is this logic supplemental to the game over state?

**Comment by GEvanscsumb** [comment]
Yes, the functionality from start to finish runs as expected.

File: GameEngine/CheeseRoom.py [hunk 4]
```diff
@@ -1,12 +1,54 @@
...
+                    showInformation("Dan exchanged the sword for the cheese. He's not an expert in cheeses, but he thinks it'd listen to RnBrie.")
+                    player.strength -= 40
+                else:
+                    return
+            elif "cheese" in player.inventory:
+                if self.checkAnswer(requestString("Exchange the cheese for the sword? (y/n)")):
+                    player.inventory.remove("cheese")
+                    player.inventory.append("sword")
+                    showInformation("Dan exchanged the cheese for the sword. What would a piece of cheese do to a rat anyway?.")
+                    player.strength += 40
```
Comment (tramstheman) [inline GameEngine/CheeseRoom.py]: this seems like a rather large strength modifier. How drastically does it affect gameplay?

File: GameEngine/BossRoom.py (hunk 1 above)
Comment (tramstheman) [inline GameEngine/BossRoom.py]: @GEvanscsumb can you see it?

File: GameEngine/Utils.py (hunk 3 above)
Comment (GEvanscsumb) [inline GameEngine/Utils.py]: If the enemy had a higher dex stat, and Dan won the battle, he didn't gain a level with the code previously

File: GameEngine/BossRoom.py (hunk 1 above)
Comment (GEvanscsumb) [inline GameEngine/BossRoom.py]: @tramstheman yes I see it

Comment (GEvanscsumb) [inline GameEngine/BossRoom.py]: With how it is now, the boss rail jumps straight into dialogue from Mark which is combined with the description of the room in "preBoss.txt". We could separate the description from the dialogue to make the popups flow better.

File: GameEngine/CheeseRoom.py (hunk 4 above)
Comment (GEvanscsumb) [inline GameEngine/CheeseRoom.py]: It's used to make the  2nd skeleton mini-boss encounter possible compared to the initial fight. I have the skeleton's hp set to 100 in both fights- first fight to suggest the player run away, second to keep it logically the same.

File: GameEngine/Entrance.py (hunk 2 above)
Comment (GEvanscsumb) [inline GameEngine/Entrance.py]: There would be random rolls where the player couldn't run from the skeleton. Though it's not a huge deal, running from the fight is the only way to continue with the story. The skeleton fight is followed by a rat fight and I didn't want to punish the user too much.

Comment (tramstheman) [inline GameEngine/Entrance.py]: that makes sense

File: GameEngine/Utils.py (hunk 3 above)
Comment (tramstheman) [inline GameEngine/Utils.py]: gotcha

File: GameEngine/CheeseRoom.py (hunk 4 above)
Comment (tramstheman) [inline GameEngine/CheeseRoom.py]: makes sense

**Pull request merged**